### ISSUE TITLE
module restore is not purging correctly

### ISSUE THREAD

**Issue by rtmclay** [TACC/Lmod#717]
When restoring a collection, the previously loaded module are not being purged correctly.

**Comment by rtmclay** [comment]
Hey Matt.  I was able to reproduce your problem and it was a bug in Lmod.  The issue was that the previously loaded modules were purged. However the changes to the environment were lost when the collection was being restored.

I have uploaded to github the branch named IS717-restore.  Please test this branch out to see if it solves the issue for you.

**Comment by mathomp4** [comment]
@rtmclay Sorry for the wait. My sysadmins got lmod-devel on my node and, huzzah!, it works!

```
> ml purge
> ml restore
Restoring modules from user's default
> echo $FC

> ml gcc
> echo $FC
/ford1/share/gmao_SIteam/gcc/gcc-13.3.0/bin/gfortran
> ml restore
Restoring modules from user's default
> echo $FC

```

**Comment by rtmclay** [comment]
Great.  Thanks for working with me to fix this bug!  Closing this issue.

**Comment by rtmclay** [comment]
Note that this branch has been merged onto the main branch.  It has been released as Lmod 8.7.48.
